openghg/util/_time.py

```diff
@@ -911,3 +913,51 @@ def dates_in_range(
             in_date.append(key)
 
     return in_date
+
+
+def evaluate_sampling_period(sampling_period: Optional[Union[Timedelta, str]]) -> Optional[str]:
+    """
+    Check the sampling period input and convert this into a string containing the
+    sampling period in seconds.
+
+    Args:
+        sampling_period: str or Timedelta value for the time to sample.
+
+    Returns:
+        str : Sampling period as a string containing the number of seconds.
+
+    TODO: Integrate sampling_period handling into logic for time_period (if practical)
+    """
+    # If we have a sampling period passed we want the number of seconds
+    if sampling_period is not None:
+
+        # Check value passed is not just a number with no units
+        try:
+            float(sampling_period)
+        except (ValueError, TypeError):
+            # If this cannot be evaluated to a float assume this is correct form.
+            pass
+        else:
+            raise ValueError(
+                f"Invalid sampling period: '{sampling_period}'. Must be specified as a string with unit (e.g. 1m for 1 minute)."
+            )
+
+        # Check string passed can be evaluated as a Timedelta object and extract this in seconds.
+        try:
+            sampling_period_td = Timedelta(sampling_period)
+        except ValueError:
+            raise ValueError(
+                f"Could not evaluate sampling period: '{sampling_period}'. Must be specified as a string with valid unit (e.g. 1m for 1 minute)."
+            )
```

Comment:
```suggestion
            ) from e
```
This should make the error message more specific (it will say that the pandas `ValueError` directly caused the error that we raise).